frontend-mini-challenge/components/LoggedInPage.tsx

```diff
@@ -131,13 +152,14 @@ Then when you make use of this page component:
     </Page>
 
 */
-export async function getServerSidePropsForLoggedInPage({ query, req }) {
-  const whoAmIResponse = await serverSideWhoAmI({ query, req });
+
+export async function getServerSidePropsForLoggedInPage({ query, req }: { query: { [key: string]: string }, req: IncomingMessage }) {
```

Comment:
There is no reason to invent things here. Look:

```
type GetServerSidePropsContext<
  Q extends ParsedUrlQuery = ParsedUrlQuery
> = {
  req: IncomingMessage & { 
    cookies: NextApiRequestCookies
  }
  res: ServerResponse
  params?: Q
  query: ParsedUrlQuery
  preview?: boolean
  previewData?: PreviewData
  resolvedUrl: string
  locale?: string
  locales?: string[]
  defaultLocale?: string
}
```
This information comes from https://github.com/vercel/next.js/blob/v10.2.0/packages/next/types/index.d.ts#L136-L151 



What datatype is query? What datatype is req?  Please use those. We know where these props come from, we know exactly what the types are.